### PR TITLE
Somewhat preserve the input graph's edge order 

### PR DESCRIPTION
**Pull request by uruuru**
Try to preserve the input graph's edge order up to the point where the internal graph is passed to the crossing minimization. The goal is that (if it doesn't matter from the crossing minimization's perspective) the drawing reflects the edge order. 

For this, the graph import had to be altered to iterate `#getContainedEdges` instead of each node's outgoing edges. 

Please double-check the inside self loop handling.

### CODE REVIEW

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PortListSorter.java [hunk 1]
```diff
@@ -1,135 +1,264 @@
...
- * @author uru
- * @kieler.design 2012-08-10 chsch grh
- * @kieler.rating proposed yellow by msp
  */
 public final class PortListSorter implements ILayoutProcessor<LGraph> {
 
     /**
-     * A comparer for ports. Ports are sorted by side (north, east, south, west) in clockwise order,
-     * beginning at the top left corner.
+     * {@inheritDoc}
```
Comment (le-cds) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PortListSorter.java]: I thought we voted against using `{@inheritDoc}`?

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/options/PortSortingStrategy.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.eclipse.elk.core.options.PortSide;
+
+/**
+ * Strategy with which a node's ports are ordered by the {@link PortListSorter}. Only relevant if the node's
+ * {@link PortConstraints} are {@link PortConstraints#FIXED_SIDE}.
+ */
+public enum PortSortingStrategy {
+
+    /**
+     * Tries to preserve the input order. That is, if a pair of ports
```
Comment (le-cds) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/options/PortSortingStrategy.java]: ...?

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PortListSorter.java [hunk 3]
```diff
@@ -1,135 +1,264 @@
...
+        int n = (highIdx - lowIdx) / 2;
+        for (int i = 0; i < n; ++i) {
+            LPort tmp = ports.get(lowIdx + i);
+            ports.set(lowIdx + i, ports.get(highIdx - i - 1));
+            ports.set(highIdx - i - 1, tmp);
+        }
+    }
+    
+        
+    private static final Function<LPort, Iterable<LEdge>> INC_EDGES = p -> p.getIncomingEdges();
```
Comment (le-cds) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PortListSorter.java]: `IN_EDGES` would sound less as if that thing would increment things.

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/transform/ElkGraphImporter.java [hunk 4]
```diff
@@ -208,48 +208,45 @@ private void importFlatGraph(final ElkNode elkgraph, final LGraph lgraph) {
...
+            } 
+        }
+
+        // now collect inside self loops of 'elkgraph'
+        if (elkgraph.getParent() != null) {
+            for (ElkEdge elkedge : elkgraph.getParent().getContainedEdges()) {
+                ElkNode source = ElkGraphUtil.getSourceNode(elkedge);
+                if (source == elkgraph && elkedge.isSelfloop()) {
+                    boolean enableInsideSelfLoops = source.getProperty(LayeredOptions.INSIDE_SELF_LOOPS_ACTIVATE);
+                    boolean isInsideSelfLoop = enableInsideSelfLoops && elkedge.isSelfloop()
```
Comment (le-cds) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/transform/ElkGraphImporter.java]: We already know that the edge is a self loop (see line 239).

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PortListSorter.java (hunk 1 above)
Comment (uruuru) [reply plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PortListSorter.java]: This is just an unfortunate diff. The inheritDoc has been there before. 

I'm completely sure regarding our decision (which we probably should put into the documentation).

**Pull request merged**